### PR TITLE
Upgrade versions in CI matrix and fix CI issue

### PR DESCRIPTION
**Pull request by sayanshaw24**

### CODE REVIEW

**Comment by wenbingl** [comment]
I don't think a lot of packages are available for Python 3.13, Can we just shift ORT version only?

**Review by wenbingl — COMMENTED** [review]
We cannot drop python 3.8 support. and should only shift ort version from [1.13, 1.17] to [1.15, 1.19]

**Comment by edgchen1** [comment]
> We cannot drop python 3.8 support. and should only shift ort version from [1.13, 1.17] to [1.15, 1.19]

FYI, Python 3.8 is EOL now. https://devguide.python.org/versions/

do we still need to support it?

**Comment by wenbingl** [comment]
> > We cannot drop python 3.8 support. and should only shift ort version from [1.13, 1.17] to [1.15, 1.19]
> 
> FYI, Python 3.8 is EOL now. https://devguide.python.org/versions/
> 
> do we still need to support it?

I just happened to see the HF's transformers Python package 4.46 yanked because it works with Python 3.8 https://pypi.org/project/transformers/4.46.0/. At least, it should be dropped after Python 3.13 was fully supported.

**Comment by wenbingl** [comment]
> > > We cannot drop python 3.8 support. and should only shift ort version from [1.13, 1.17] to [1.15, 1.19]
> > 
> > 
> > FYI, Python 3.8 is EOL now. https://devguide.python.org/versions/
> > do we still need to support it?
> 
> I just happened to see the HF's transformers Python package 4.46 yanked because it works with Python 3.8 https://pypi.org/project/transformers/4.46.0/. At least, it should be dropped after Python 3.13 was fully supported.

it seems ORT dropped, we can drop it too.

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -445,26 +445,26 @@ stages:
...
+          ort.version: '1.16.3'
+        py39-1151:
           python.version: '3.9'
-          torch.version: 'torch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 --index-url https://download.pytorch.org/whl/cpu'
-          ort.version: '1.14.1'
-        py38-1131:
-          python.version: '3.8'
-          torch.version: 'torch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 -f https://download.pytorch.org/whl/torch_stable.html'
-          ort.version: '1.13.1'
+          torch.version: 'torch torchvision torchaudio'
```
Comment (wenbingl) [inline .pipelines/ci.yml:466]: can you keep the torch version for some backward compatibility?

**Review by wenbingl — COMMENTED** [review]
Can you also the build configuration for MacOS and Linux too?

File: .pipelines/ci.yml (hunk 1 above)
Comment (sayanshaw24) [reply .pipelines/ci.yml]: done!

**Review by wenbingl — APPROVED** [review]

**Pull request merged**